operator/cnp_event.go

```diff
@@ -91,11 +132,50 @@ func enableCNPWatcher() error {
 	controller.NewManager().UpdateController("cnp-to-groups",
 		controller.ControllerParams{
 			DoFunc: func(ctx context.Context) error {
-				groups.UpdateCNPInformation()
+				// Retrieves all the CNP that has currently a
+				// derivative policy and creates the new
+				// derivatives policies with the latest
+				// information from providers.  To avoid issues
+				// with rate-limiting this function will
+				// execute the addDerivative function with a
+				// max number of concurrent calls, defined on
+				// maxConcurrentUpdates.
+				cnpToUpdate := cnpCache.GetAllCNP()
+				sem := make(chan bool, maxConcurrentUpdates)
+				for _, cnp := range cnpToUpdate {
+					sem <- true
+					go func(cnp *cilium_v2.CiliumNetworkPolicy) {
+						defer func() { <-sem }()
+						addDerivativeCNP(cnp)
```

Comment:
100 CNP updates in a single burst will probably DDoS a kube-apiserver. I don't see the point of doing this concurrently, as these policies are eventually going to be updated with the latest information from provider in k8s. The "eventually" is the keyword here, what makes it better to do it concurrently?

Reply:
This is rate limited by `sem` channel with `maxConcurrentUpdates` size.